var/spack/repos/builtin/packages/lua/package.py

```diff
@@ -58,24 +58,33 @@ class Lua(Package):
         destination="luarocks",
         placement='luarocks')
 
+    patch('https://git.archlinux.org/svntogit/packages.git/plain/trunk/liblua.so.patch?h=packages/lua&id=70abed4cfa74993348bf7afbbc70151a3d57cc01',
+            sha256='2cc83c77423a2dda3696766b2d1ccee2796e052ab04d5178905f41ed9241a3d8')
+
     def install(self, spec, prefix):
         if spec.satisfies("platform=darwin"):
             target = 'macosx'
         else:
             target = 'linux'
         make('INSTALL_TOP=%s' % prefix,
+             'MYCFLAGS=%s' % (
+                 self.compiler.pic_flag),
              'MYLDFLAGS=-L%s -L%s' % (
                  spec['readline'].prefix.lib,
                  spec['ncurses'].prefix.lib),
              'MYLIBS=-lncursesw',
              'CC=%s -std=gnu99' % spack_cc,
              target)
         make('INSTALL_TOP=%s' % prefix,
+             'MYCFLAGS=%s' % (
+                 self.compiler.pic_flag),
              'MYLDFLAGS=-L%s -L%s' % (
                  spec['readline'].prefix.lib,
                  spec['ncurses'].prefix.lib),
              'MYLIBS=-lncursesw',
              'CC=%s -std=gnu99' % spack_cc,
+             'TO_LIB=liblua.a liblua.so liblua.so.%s liblua.so.%s' % (
```

Comment:
sorry, another issue: this will certainly fail on macOS. Can you please get the list dynamically?

Reply:
ah, I see that your patch won't work on macOS either, maybe don't apply it on darwin and skip this line as well?

Reply:
on macOS that would be
```
TO_LIB=liblua.a liblua.dylib
```